autoPyTorch/pipeline/image_classification.py

```diff
@@ -0,0 +1,206 @@
+
+                return y
+
+    def _get_hyperparameter_search_space(self,
+                                         dataset_properties: Dict[str, Any],
+                                         include: Optional[Dict[str, Any]] = None,
+                                         exclude: Optional[Dict[str, Any]] = None,
+                                         ) -> ConfigurationSpace:
+        """Create the hyperparameter configuration space.
+
+        For the given steps, and the Choices within that steps,
+        this procedure returns a configuration space object to
+        explore.
+
+        Args:
+            include (Optional[Dict[str, Any]]): what hyper-parameter configurations
+                to honor when creating the configuration space
+            exclude (Optional[Dict[str, Any]]): what hyper-parameter configurations
+                to remove from the configuration space
+            dataset_properties (Optional[Dict[str, Union[str, int]]]): Caracteristics
+                of the dataset to guide the pipeline choices of components
+
+        Returns:
+            cs (Configuration): The configuration space describing
+                the SimpleRegressionClassifier.
+        """
+        cs = ConfigurationSpace()
+
+        if dataset_properties is None or not isinstance(dataset_properties, dict):
+            dataset_properties = dict()
+        if 'target_type' not in dataset_properties:
+            dataset_properties['target_type'] = 'image_classification'
+        if dataset_properties['target_type'] != 'image_classification':
+            dataset_properties['target_type'] = 'image_classification'
+        # get the base search space given this
+        # dataset properties. Then overwrite with custom
+        # classification requirements
+        cs = self._get_base_search_space(
+            cs=cs, dataset_properties=dataset_properties,
+            exclude=exclude, include=include, pipeline=self.steps)
+
+        # Here we add custom code, like this with this
+        # is not a valid configuration
+
+        self.configuration_space = cs
+        self.dataset_properties = dataset_properties
+        return cs
+
+    def _get_pipeline_steps(self, dataset_properties: Optional[Dict[str, Any]],
+                            ) -> List[Tuple[str, autoPyTorchChoice]]:
+        """
+        Defines what steps a pipeline should follow.
+        The step itself has choices given via autoPyTorchChoice.
+
+        Returns:
+            List[Tuple[str, autoPyTorchChoice]]: list of steps sequentially exercised
+                by the pipeline.
+        """
+        steps = []  # type: List[Tuple[str, autoPyTorchChoice]]
+
+        default_dataset_properties = {'target_type': 'image_classification'}
+        if dataset_properties is not None:
+            default_dataset_properties.update(dataset_properties)
+
+        steps.extend([
+            ("pad", PadChoice(default_dataset_properties)),
```

Comment:
I think probably padding makes sense as augmentation?

Reply:
okay then I'll move it to augmentation when I make a PR for that.